sys/riotboot/serial.c

```diff
@@ -53,6 +55,20 @@ static inline void uart_write_byte(uart_t uart, uint8_t data)
     uart_write(uart, &data, 1);
 }
 
+static inline bool _boot_pin(void)
```

Comment:
This (and the later GPIO init) is duplicating _bootloader_alternative_mode of `bootloaders/riotboot_dfu/main.c`. I don't care which version survives, but have a slight preference to only have one. (Especially as they are imlpemented slightly differently).

Reply:
I mean it's just `gpio_init()` and `gpio_read()`, I don't think the duplication of those two lines is bad.
`riotboot_serial` has a sync phase, so it allows to press the button at any time before the timeout occurs, it's not required to have the button pressed on reset.

Reply:
Fair enough. I guess the better place to deduplicate here is to make these preprocessor macros into something more init-and-get. (I just don't like how the evaluation logic of the pin inversion is done slightly differently in two places).